feat: code tabs and examples for login checks (#1125)
ory/docs: docs/identities/sign-in/check-session.mdx

---
id: check-session

> title: Check for user login sessions in browser cookies and session tokens

Reviewer: ```suggestion
title: Check for user session status in browser cookies and session tokens
```

we check the session status not the login status, right? 
So checking login status would be e.g. checking the status of the login flow, 
we check if the session has status "active" or not. 
At least that is how I would understand it.

sidebar_label: Check login session
slug: check-session-token-cookie-api
---

> # Check login status

Reviewer: ```suggestion
# Check login session
```
or

```suggestion
# Check session status
```

import CodeBlock from '@theme/CodeBlock'
import CodeFromRemote from '@theme/CodeFromRemote'
import expressjs from '!!raw-loader!../../../code-examples/identity/sign-in/check-session/js/server.js'
import ajax from '!!raw-loader!../../../code-examples/identity/sign-in/check-session/js/client.js'
import goExample from '!!raw-loader!../../../code-examples/identity/sign-in/check-session/go/server.go'

<Tabs>
  <TabItem value="curl" label="curl" default>

To check for a session token, use:

```shell
curl -H "Authorization: Bearer {your-session-token}" \
  "https://{project.slug}.projects.oryapis.com/sessions/whoami"
```

To check for a session cookie, use:

```shell
curl -H "Cookie: ory_session_...=..." \
  "https://{project.slug}.projects.oryapis.com/sessions/whoami"
```

  </TabItem>
  <TabItem value="ajax" label="AJAX / SPA" >
    <p>When calling Ory or your API from a Single Page Application using AJAX, you need to ensure three things:</p>
    <ol>
      <li>The Single Page App (<code>www.yourdomain.com</code>, <code>localhost:3000</code>), Ory (<code>ory.yourdomain.com</code>, <code>localhost:4000</code>), and your API are all on the domain.</li>

>       <li>You configured CORS corretly in the Ory Console and your API server.</li>

Reviewer: ```suggestion
      <li>You configured CORS in the Ory Console and your API server.</li>
```

no point in configuring it incorrectly ;-)

To check for an active session with an Ory Session Cookie, run:

```shell
curl -H "Cookie: ory_session_...=..." \
  "https://{project.slug}.projects.oryapis.com/sessions/whoami"
```

</TabItem>
<TabItem value="ajax" label="AJAX / SPA">

When calling Ory or your API from a Single Page Application (SPA) using AJAX:

- The SPA (`www.yourdomain.com`, `localhost:3000`), Ory (`ory.yourdomain.com`, `localhost:4000`), and your API must be on the same domain.
- You configured CORS in the Ory Console and on your API server.
- The cookie is configured to be available on the domains you need, for example, the full `yourdomain.com`.

<CodeBlock className="language-js">{ajax}</CodeBlock>

</TabItem>
<TabItem value="expressjs" label="ExpressJS">

Ory checks for both the cookie and the token, which means that your API must handle both Ory Session Cookies and Ory Session Tokens:

<CodeBlock className="language-js">{expressjs}</CodeBlock>

</TabItem>
<TabItem value="go" label="Go">

Ory checks for both the cookie and the token, which means that your API must handle both Ory Session Cookies and Ory Session Tokens:

<CodeBlock className="language-go">{goExample}</CodeBlock>

</TabItem>
</Tabs>
````

> ## Session response

Reviewer 2: There's more about sessions (including a full session example with some explainers) here: 
https://www.ory.sh/docs/kratos/session-management/overview#ory-session

Maybe instead of duplicating this content, it'd be better to point the user toward this set of docs?

Author: Hm we could probably create a file and import it then across the different pages. Since the sessions section has not been touched yet, i would probably keep it here for now, but we can have a follow up issue to clean it up!
